### ISSUE TITLE
File > Open > open a folder does not work on the first attempt

### ISSUE THREAD

**Issue by kevin-lin** [Microsoft/vscode#10317]
- VSCode Version: 1.4.0
- OS Version: OSX 10.11.6

Steps to Reproduce:

1. Open new VSCode window
2. In the menu bar, File > Open > select a folder

Every time I try to open a folder using this method, I need to repeat the steps twice in order for the folder to open in my VSCode editor.

**Comment by bpasero** [comment]
I am very sorry to hear you are having issues with VS Code. I wonder if this issue is being caused by an installed extension. Can you try to run VS Code without extensions? From the command line, execute: `code --disable-extensions` and try your steps again to see if it reproduces. If you see it is an issue with the extension, please file it against the extension repository itself.

**Comment by bpasero** [comment]
Are you possibly running Code as admin maybe?

**Comment by kevin-lin** [comment]
can still repro with extensions disabled.

**Comment by LordZane** [comment]
I have the same issue on version 1.4.0 windows 10 build 14393, and it happens with or without admin

**Comment by bpasero** [comment]
@kevin-lin @LordZane can you first bring up the developer console (from the help menu) and then try to open the folder to see if there is an error logged? if nothing is logged, can you run "code --verbose" from the command line, do the opening of a folder and see if there is output printed to the command line?

**Comment by LordZane** [comment]
Its weird, now the first folder I open fails, but every folder after that works. Before it would only open every other one, so I had to hit any folder twice I wanted to open if that makes sense. I do get this error though:
`EPERM: operation not permitted, open 'C:\Users\zane\AppData\Roaming\Code\storage.json': Error: EPERM: operation not permitted, open 'C:\Users\zane\AppData\Roaming\Code\storage.json'
    at Error (native)
    at Object.fs.openSync (original-fs.js:584:18)
    at Object.fs.writeFileSync (original-fs.js:1234:33)
    at e.save (C:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\code\electron-main\main.js:8:2569)
    at e.setItem (C:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\code\electron-main\main.js:8:2123)
    at C:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\code\electron-main\main.js:9:17765
    at Function.wrappedCallback (C:\Program Files (x86)\Microsoft VS Code\resources\electron.asar\browser\api\dialog.js:89:14)`

**Comment by bpasero** [comment]
Looks like you opened Code once as administrator and that might have caused the storage.json file to be created with admin rights. In general it is not recommended to start an application as administrator because subsequent starts as normal user can fail.

**Comment by kevin-lin** [comment]
I got a similar error to Zane on OSX
```
shell.ts:422 EACCES: permission denied, open '/Users/<user>/Library/Application Support/Code/storage.json': Error: EACCES: permission denied, open '/Users/<user>/Library/Application Support/Code/storage.json'
    at Error (native)
    at Object.fs.openSync (original-fs.js:584:18)
    at Object.fs.writeFileSync (original-fs.js:1234:33)
    at e.save (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-main/main.js:8:2569)
    at e.setItem (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-main/main.js:8:2123)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-main/main.js:9:17765
    at Function.wrappedCallback (/Applications/Visual Studio Code.app/Contents/Resources/electron.asar/browser/api/dialog.js:89:14)

e.onUnexpectedError @ shell.ts:422
(anonymous function) @ shell.ts:324
e.onUnexpectedError @ errors.ts:73
u @ errors.ts:88
(anonymous function) @ integration.ts:107
emitTwo @ events.js:100
emit @ events.js:185
```

**Comment by bpasero** [comment]
We fixed this particular issue for insiders and it will be available for 1.5 release.

Both of you should however make sure that the user you run Code with has sufficient permissions to read and write to the folder where `storage.json` is in.

We are releasing preview releases of the next stable VS Code version for everyone to try and give feedback. These preview releases are not 100% tested and might be unstable but contain our latest features and bugfixes. You can give our preview releases a try from: http://code.visualstudio.com/Download#insiders

**Comment by aeschli** [comment]
Verified
- remove perissions to read storage.json 
- start VSCode
- VSCode now starts up without error message. No folder history, but all seems to work.

Starting `code --verbose` shows the issue

Created #11440 as I think a warning to the user would be good to have.
